### ISSUE TITLE
Workspaces, applications, laucher icons do not open on touch screen with finger select. 

### ISSUE THREAD

**Issue by math93033** [pop-os/gnome-shell-extension-pop-cosmic#376]
**Distribution (run `cat /etc/os-release`):**
 NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os

 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

**Issue/Bug Description:**
I am on a Microsoft Surface Pro 6 device. I have installed the surface specific kernel from the suface-linux project. All touch screen functions of the tablet/computer work correctly. However, I am unable to open the Pop_OS specific UI launchers (Applications, workspaces, and search) from the dock or from the top bar. Specifically, I cannot open them with finger touch, but  they work correctly when selected with a mouse or touchpad. 

**Steps to reproduce (if you know):**

**Expected behavior:**
They icons do not work when touched with a finger unless the touch is held for a few seconds and then the windows close or seem to malfunction. 

**Other Notes:**

**Comment by jacobgkau** [comment]
Not sure this is cosmic-dock if it's also the top bar, could be the main COSMIC extension.

**Comment by jacobgkau** [comment]
@math93033 I assume from your description that other apps open from the dock normally?

**Comment by leviport** [comment]
I tested on a different model of laptop with a touch screen. I can kinda get them to work, but they're definitely inconsistent. Applications in particular seems iffy. I see it pop up briefly, then go away. I'm not sure if it's just me, but it seems like they work better on an empty workspace, with no windows already open. Any app that also summons the on-screen keyboard seems to make the behavior even more strange.

It seems like the Workspaces and Applications buttons on the top bar are a lot more reliable, but the App Library does sometimes disappear on me. It could definitely be the main extension though.

**Comment by math93033** [comment]
@leviport @jacobgkau   Yes. Everything else opens correctly. It's only the pop specific dock and top bar navigation icons that do not work.

**Comment by leviport** [comment]
Yeah it's probably not specific to the dock then. Moved to pop-os/cosmic repo.

**Comment by CR0CKER** [comment]
Can confirm this issue also exists on Surface Go 2 with latest PopOS. Both the top bar and the dock buttons do not respond to tapping on the touch screen.
